opendevin/llm/llm.py

```diff
@@ -1,16 +1,33 @@
+import os
+import uuid
+
 from litellm import completion as litellm_completion
 from functools import partial
-import os
 
 DEFAULT_MODEL = os.getenv("LLM_MODEL", "gpt-4-0125-preview")
 DEFAULT_API_KEY = os.getenv("LLM_API_KEY")
+PROMPT_DEBUG_DIR = os.getenv("PROMPT_DEBUG_DIR", "")
 
 class LLM:
-    def __init__(self, model=DEFAULT_MODEL, api_key=DEFAULT_API_KEY):
+    def __init__(self, model=DEFAULT_MODEL, api_key=DEFAULT_API_KEY, debug_dir=PROMPT_DEBUG_DIR):
         self.model = model if model else DEFAULT_MODEL
         self.api_key = api_key if api_key else DEFAULT_API_KEY
-
-        self._completion = partial(litellm_completion, model=self.model, api_key=self.api_key)
+        self._debug_dir = debug_dir
+        self._debug_idx = 0
+        self._debug_id = uuid.uuid4().hex
+        if self._debug_dir:
+            print(f"Logging prompts to {self._debug_dir}/{self._debug_id}")
+        self._completion_unwrapped = partial(litellm_completion, model=self.model, api_key=self.api_key)
+        def wrapper(*args, **kwargs):
+            if "messages" in kwargs:
+                messages = kwargs["messages"]
+            else:
+                messages = args[1]
+            resp = self._completion_unwrapped(*args, **kwargs)
+            message_back = resp['choices'][0]['message']['content']
+            self.write_debug(messages, message_back)
+            return resp
+        self._completion = wrapper
```

Comment:
I guess by wrapping the `._completion` function, we will break the documentation and argument hints

Reply:
I think that that should make it work? since the wrapping is only happening at the runtime, and the argument hints only depends on static type?

Reply:
👍 changed!